src/compiler/checker.ts

```diff
@@ -167,10 +167,10 @@ namespace ts {
 
         return checker;
 
-        function getEmitResolver(sourceFile?: SourceFile) {
+        function getEmitResolver(sourceFile: SourceFile, cancellationToken: CancellationToken) {

```

Comment:
I guess i'm not really understanding the complaint.

The reason the cancellation token works as a singleton elsewhere is because it truly operates as a singleton around all the operations it wraps.  That's not the case for the typechecker.  For some codepaths cancellation should be respected.  Through others it should be ignored.  The easiest (and most natural IMO) way to encode that is to actually pass the cancellation token (or not) through those codepaths.  

So... i agree that the issue with the checker is "an implementation detail we should look into solving".  It's just that the best solution (IMO) was to follow this pattern of cancellation inside of it :)